pkg/otelcollector/infra-meter/infra-meter.go

```diff
@@ -143,13 +214,24 @@ func mapSlice(mapping map[string]string, xs []string) []string {
 	return ys
 }
 
-// normalizeComponentName normalizes user defines component name by adding
-// `user-defined-<pipeline_name>` suffix.
-// This ensures no builtin components are overwritten.
-func normalizeComponentName(pipelineName, componentName string) string {
-	suffix := fmt.Sprintf("user-defined-%s", pipelineName)
-	if componentName == "" {
-		return suffix
+func updateK8sAttributesProcessor(processor interface{}, selectorsList []interface{}) map[string]interface{} {
```

Comment:
Nit: no need to return a new map, you're modifying it in place anyway

Reply:
But we don't take the map as input and it is initialised in the function only

Reply:
we take the processor from `config.Processors[strID]`, from which we get the map. which we edit in place, right?

Reply:
We get interface from config.Processors[strID] and then type cast it to map

Reply:
Made the changes. Thank you!